backend/api/spec.yaml

```
@@ -0,0 +1,1889 @@
+        "404":
+          description: App not found response
+        "500":
+          description: Get App error response
+    put:
+      description: update app
+      operationId: updateApp
+      security:
+        - oidcBearerAuth: []
+        - oidcCookieAuth: []
+        - githubCookieAuth: []
+      parameters:
+        - in: path
+          name: appID
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: payload for update app
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/appConfig"
+      responses:
+        "200":
+          description: Update App success response
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/application"
+        "400":
+          description: Bad request response
+        "500":
+          description: Update app error response
+    delete:
+      description: delete app
+      operationId: deleteApp
+      security:
+        - oidcBearerAuth: []
+        - oidcCookieAuth: []
+        - githubCookieAuth: []
+      parameters:
+        - in: path
+          name: appID
+          required: true
+          schema:
+            type: string
+      responses:
+        "204":
+          description: Delete App success response
+        "500":
+          description: Delete App error response
+
+  /api/apps/{appID}/groups:
+    get:
+      description: paginate groups of an app
+      operationId: paginateGroups
+      security:
+        - oidcBearerAuth: []
+        - oidcCookieAuth: []
+        - githubCookieAuth: []
+      parameters:
+        - in: path
+          name: appID
+          required: true
+          schema:
+            type: string
+        - in: query
+          name: page
+          required: false
+          schema:
+            type: integer
+            minimum: 0
```

This should have a `default: 1`.

---

The frontend doesn't send the `page` and `perpage` query params so it is marked `required: false` and Internally have handled the value to default to `1` if `page` is not passed.

---

From what you say, it sounds like this defaults to 1? Which means we should mark it in the spec as default 1... unless I'm not understanding something.

---

The generated server code doesn't default the value to 1 if it is not provided in the request that is why it was handled this way manually.

---

Thanks. I'll look into this. Maybe we need to raise an issue about it not setting the default.

---

Moved this into an issue and out of this PR instead: https://github.com/kinvolk/nebraska/issues/535